src/tamp_improv/approaches/improvisational/policies/rl2mpc.py

```diff
@@ -0,0 +1,196 @@
+"""RL2MPC policy implementation combining RL and MPC."""
+
+from copy import deepcopy
+from dataclasses import dataclass, field
+from typing import cast
+
+import gymnasium as gym
+import numpy as np
+
+from tamp_improv.approaches.improvisational.policies.base import (
+    ActType,
+    ObsType,
+    Policy,
+    PolicyContext,
+    TrainingData,
+)
+from tamp_improv.approaches.improvisational.policies.mpc import MPCConfig, MPCPolicy
+from tamp_improv.approaches.improvisational.policies.rl import (
+    RLConfig,
+    RLPolicy,
+    TrainingProgressCallback,
+)
+
+
+@dataclass
+class RL2MPCConfig:
+    """Configuration for RL2MPC policy."""
+
+    rl_config: RLConfig = field(default_factory=RLConfig)
+    mpc_config: MPCConfig = field(default_factory=MPCConfig)
+    reward_threshold: float = -5.0  # Threshold to switch from RL to MPC
+    window_size: int = 50  # Window size for computing average reward
+
+
+class RL2MPCCallback(TrainingProgressCallback):
+    """Callback that stops training when reward threshold is reached."""
+
+    def __init__(
+        self,
+        reward_threshold: float,
+        window_size: int = 50,
+        check_freq: int = 100,
+        verbose: int = 1,
+    ):
+        super().__init__(check_freq=check_freq, verbose=verbose)
+        self.reward_threshold = reward_threshold
+        self.window_size = window_size
+        self.threshold_reached = False
+
+    def _on_step(self) -> bool:
+        should_continue = super()._on_step()
+
+        # Check threshold on episode completion
+        if len(self.episode_rewards) >= self.window_size:
+            recent_mean = np.mean(self.episode_rewards[-self.window_size :])
```

Comment:
I just want to call attention to the fact that we're checking step-wise rewards instead of episode-wise returns (cumulative rewards). in general we might care more about returns than rewards. for example, if we have a choice between two episodes with rewards:
1. `0, 0, 0, 0, 0, 0, 0, 0, 0, 10, [terminated]`
2. `0, 2.5, [terminated]`

then maybe we should prefer (1), but mean-rewards would have us prefer (2)? I'm not sure if this intuition applies here, though. what do you think?